getting-started/quickstart-guides/react.md

```diff
@@ -789,6 +791,129 @@ Fetch uploaded image and show in UI using `IKImage` with the `filePath` returned
 
 The app should display your uploaded image correctly!
 
+## **Advanced file upload**
+
+A more detailed example for how to use the file upload component is presented below:
+
+{% tabs %}
+{% tab title="React JSX" %}
+{% code title="src/App.js" %}
+```jsx
+import React, { useRef } from 'react';
+import './App.css';
+import { IKContext, IKImage, IKUpload } from 'imagekitio-react';
+
+const publicKey = '<YOUR_IMAGEKIT_PUBLIC_KEY>';
+const urlEndpoint = '<YOUR_IMAGEKIT_URL_ENDPOINT>';
+const authenticationEndpoint = 'http://localhost:3001/auth';
+
+const onError = err => {
+  console.log("Error", err);
+};
+
+const onSuccess = res => {
+  console.log("Success", res);
+};
+
+const onUploadProgress = progress => {
+  console.log("Progress", progress);
+};
+
+const onUploadStart = evt => {
+  console.log("Start", evt);
+};
+
+function App() {
+  const reftest = useRef(null);
```

Comment:
Only code examples are preset in sample app as well. This is a guide so we need to elaborate and create proper sections.